src/xtgeo/grid3d/_roff_grid.py

```diff
@@ -334,13 +344,17 @@ def from_xtgeo_grid(xtgeo_grid):
 
         return RoffGrid(nx, ny, nz, corner_lines, zvals, split_enz, active, subgrids)
 
-    def to_file(self, filelike, roff_format=roffio.Format.BINARY):
+    def to_file(
+        self,
+        filelike: Union[str, pathlib.Path, IO],
+        roff_format: roffio.Format = roffio.Format.BINARY,
+    ) -> None:
         """
         Writes the RoffGrid to a roff file
         Args:
             filelike (str or byte stream): The file to write to.
         """
-        data = {
+        data: dict[str, dict] = {
```

Comment:
`Dict` is also used imported and capitalized elsewhere in the code. Maybe we should discuss which way to go. I am not sure if there are drawbacks while still support Python 3.8